vm/src/value.cpp

```diff
@@ -0,0 +1,533 @@
+#include "value.h"
+#include "deserializer.h"
+#include "matrix_3.h"
+#include "matrix_4.h"
+#include "serializer.h"
+#include "vector_2.h"
+#include "vector_3.h"
+#include <cassert>
+#include <utility>
+
+namespace Borges {
+
+Value::Value(bool b): type_(Type::boolean), boolean_(b)
+{
+}
+
+Value::Value(double x) : type_(Type::number), number_(x)
+{
+}
+
+Value::Value(const Vector_2& v)
+    : type_(Type::vector_2),
+      vector_2_(std::unique_ptr<const Vector_2>(new Vector_2(v)))
+{
```

Comment:
using new in member initialization is imo not a good idea. if construction fails because of memory allocation error and many members were initialized in this way, it is impossible to find which member caused the problem, and the resources for the members are leaked. furthermore, it makes it unclear which members need to be freed. However, since the new is within the construction of the unique_ptr the latter argument may not be valid, i'm not sure on this.